core/chains/evm/logpoller/orm.go

```diff
@@ -681,6 +698,20 @@ func (o *DbORM) SelectIndexedLogsWithSigsExcluding(sigA, sigB common.Hash, topic
 	return logs, nil
 }
 
+func (o *DbORM) InsertLogsWithBlock(logs []Log, block LogPollerBlock, qopts ...pg.QOpt) error {
+	// Optimization, don't open TX when there is only block to be persisted
+	if len(logs) == 0 {
```

Comment:
AFAIK, opening TX requires an additional round trip to the database. This could be a nice improvement for PollAndSave ticks when no logs are emitted